firmware/hw_layer/stepper.cpp

```diff
@@ -143,20 +143,21 @@ void StepperMotor::setDirection(bool isIncrementing) {
 
 void StepperMotor::pulse() {
 #if EFI_PROD_CODE
-	palWritePad(enablePort, enablePin, false); // enable stepper
-	palWritePad(stepPort, stepPin, true);
+	palWritePad(enablePort, enablePin, 0); // enable stepper
+	palWritePad(stepPort, stepPin, 0);
```

Comment:
that's evil to replace 'true' with '0' here! @DonaldBecker  @mck1117

Reply:
besides, what was wrong with `true` and `false`?

Reply:
@mck1117 ```#define palWritePad(port, pad, bit) pal_lld_writepad(port, pad, bit)
                                     ^~~~~~~~~~~~~~~~
./hw_layer/stepper.cpp:153:2: note: in expansion of macro 'palWritePad'
  palWritePad(enablePort, enablePin, true); // disable stepper
  ^~~~~~~~~~~
./hw_layer/stepper.cpp: In member function 'void StepperMotor::initialize(brain_pin_e, brain_pin_e, pin_output_mode_e, float, int, brain_pin_e, pin_output_mode_e, Logging*)':
ChibiOS/os/hal/ports/STM32/LLD/GPIOv2/hal_pal_lld.h:397:29: warning: '~' on an expression of type bool [-Wbool-operation]
   ((port)->BSRR.W = ((~(bits) & (mask)) << (16U + (offset))) |              \
                             ^
ChibiOS/os/hal/ports/STM32/LLD/GPIOv2/hal_pal_lld.h:425:42: note: in expansion of macro 'pal_lld_writegroup'
 #define pal_lld_writepad(port, pad, bit) pal_lld_writegroup(port, 1, pad, bit)
                                          ^~~~~~~~~~~~~~~~~~
ChibiOS/os/hal/include/hal_pal.h:592:37: note: in expansion of macro 'pal_lld_writepad'
 #define palWritePad(port, pad, bit) pal_lld_writepad(port, pad, bit)
                                     ^~~~~~~~~~~~~~~~
./hw_layer/stepper.cpp:187:2: note: in expansion of macro 'palWritePad'
  palWritePad(this->enablePort, enablePin, true); // disable stepper
  ^~~~~~~~~~~
ChibiOS/os/hal/ports/STM32/LLD/GPIOv2/hal_pal_lld.h:397:29: note: did you mean to use logical not ('!')?
   ((port)->BSRR.W = ((~(bits) & (mask)) << (16U + (offset))) |              \
                             ^
ChibiOS/os/hal/ports/STM32/LLD/GPIOv2/hal_pal_lld.h:425:42: note: in expansion of macro 'pal_lld_writegroup'
 #define pal_lld_writepad(port, pad, bit) pal_lld_writegroup(port, 1, pad, bit)
                                          ^~~~~~~~~~~~~~~~~~
ChibiOS/os/hal/include/hal_pal.h:592:37: note: in expansion of macro 'pal_lld_writepad'
 #define palWritePad(port, pad, bit) pal_lld_writepad(port, pad, bit)
                                     ^~~~~~~~~~~~~~~~
./hw_layer/stepper.cpp:187:2: note: in expansion of macro 'palWritePad'
  palWritePad(this->enablePort, enablePin, true); // disable stepper
  ^~~~~~~~~~~
ChibiOS/os/hal/ports/STM32/LLD/GPIOv2/hal_pal_lld.h:397:29: warning: '~' on an expression of type bool [-Wbool-operation]
   ((port)->BSRR.W = ((~(bits) & (mask)) << (16U + (offset))) |              \
                             ^
ChibiOS/os/hal/ports/STM32/LLD/GPIOv2/hal_pal_lld.h:425:42: note: in expansion of macro 'pal_lld_writegroup'
 #define pal_lld_writepad(port, pad, bit) pal_lld_writegroup(port, 1, pad, bit)
                                          ^~~~~~~~~~~~~~~~~~
ChibiOS/os/hal/include/hal_pal.h:592:37: note: in expansion of macro 'pal_lld_writepad'
 #define palWritePad(port, pad, bit) pal_lld_writepad(port, pad, bit)
                                     ^~~~~~~~~~~~~~~~
./hw_layer/stepper.cpp:190:2: note: in expansion of macro 'palWritePad'
  palWritePad(this->stepPort, this->stepPin, false);
  ^~~~~~~~~~~
ChibiOS/os/hal/ports/STM32/LLD/GPIOv2/hal_pal_lld.h:397:29: note: did you mean to use logical not ('!')?
   ((port)->BSRR.W = ((~(bits) & (mask)) << (16U + (offset))) |              \
                             ^
ChibiOS/os/hal/ports/STM32/LLD/GPIOv2/hal_pal_lld.h:425:42: note: in expansion of macro 'pal_lld_writegroup'
 #define pal_lld_writepad(port, pad, bit) pal_lld_writegroup(port, 1, pad, bit)
                                          ^~~~~~~~~~~~~~~~~~
ChibiOS/os/hal/include/hal_pal.h:592:37: note: in expansion of macro 'pal_lld_writepad'
 #define palWritePad(port, pad, bit) pal_lld_writepad(port, pad, bit)
                                     ^~~~~~~~~~~~~~~~
./hw_layer/stepper.cpp:190:2: note: in expansion of macro 'palWritePad'
  palWritePad(this->stepPort, this->stepPin, false);
  ^~~~~~~~~~~
./hw_layer/stepper.cpp:159:44: warning: unused parameter 'enablePinMode' [-Wunused-parameter]
   brain_pin_e enablePin, pin_output_mode_e enablePinMode, Logging *sharedLogger) {
                          ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
Compiling backup_ram.cpp
Compiling mpu_util.cpp
Compiling stm32_pins.cpp
Compiling stm32_common.cpp
Compiling cj125.cpp
Compiling cj125_logic.cpp
Compiling yaw_rate_sensor.cpp```

Reply:
ehh ok